opsdroid/events.py

```diff
@@ -286,3 +288,35 @@ class Image(File):
         """Return the image dimensions `(w,h)`."""
         fbytes = await self.get_file_bytes()
         return get_image_size_from_bytesio(io.BytesIO(fbytes), len(fbytes))
+
+
+class RoomCreation(Event):
+    """Event class to represent the creation of a new room."""
+
+    def __init__(self, public=True, *args, **kwargs):
```

Comment:
How matrix specific is the concept of public rooms? Could we have some kind of generic room properties key or something?!

Reply:
Yeah, this is a problem I just generally wasn't sure how to approach, I suspect there are going to be a lot of properties some platforms' rooms have and others' don't. Room properties makes sense, then it could just be a dict with whatever in it. 

Either way I think the event should be lenient on taking properties and it should be up to the connector to know how to deal with them.